### PR TITLE
Added IntelliCode to extensionsGallery.json

### PR DESCRIPTION
**Pull request by samoretc**
This PR adds IntelliCode to Azure Data Studio.

### CODE REVIEW

File: extensionsGallery.json [hunk 1]
```diff
@@ -63,6 +63,59 @@
 					"statistics": [],
 					"flags": "preview"
 				},
+				{
+					"extensionId": "10",
```
Comment (samoretc) [inline extensionsGallery.json]: I'm not sure what the extensionId is

File: extensionsGallery.json [hunk 2]
```diff
@@ -63,6 +63,59 @@
...
+					"extensionName": "vscodeintellicode",
+					"displayName": "Visual Studio IntelliCode",
+					"shortDescription": "AI-assisted development",
+					"publisher": {
+						"displayName": "Microsoft",
+						"publisherId": "Microsoft",
+						"publisherName": "Microsoft"
```
Comment (samoretc) [inline extensionsGallery.json]: I'm not sure if the displayName, publisherId, and publisherName should all be Microsoft. In our VSCode extension, our author's name is microsoft, but our publisher is [VisualStudioExptTeam](https://devdiv.visualstudio.com/DevDiv/_git/IntelliCode-VSCode?path=%2Fsrc%2Fms-vsintellicode%2Fpackage.json&version=GBmaster&line=6&lineStyle=plain&lineEnd=6&lineStartColumn=19&lineEndColumn=39).

File: extensionsGallery.json [hunk 3]
```diff
@@ -63,6 +63,59 @@
...
+					},
+					"versions": [
+						{
+							"version": "1.2.1",
+							"lastUpdated": "10/29/2019",
+							"assetUri": "",
+							"fallbackAssetUri": "fallbackAssetUri",
```
Comment (samoretc) [inline extensionsGallery.json]: Not 100% if these values are correct, I copied them from the agent extension.

File: extensionsGallery.json [hunk 4]
```diff
@@ -63,6 +63,59 @@
...
+									"source": "https://sqlopsextensions.blob.core.windows.net/extensions/agent/agent-0.43.0.vsix"
+								},
+								{
+									"assetType": "Microsoft.VisualStudio.Services.Icons.Default",
+									"source": "https://intellicodeextensions.blob.core.windows.net/intellicodeextensions/extensions/azuredatastudio/1.2.1/icon.png"
+								},
+								{
+									"assetType": "Microsoft.VisualStudio.Services.Content.Details",
+									"source": "https://intellicodeextensions.blob.core.windows.net/intellicodeextensions/extensions/azuredatastudio/1.2.1/README.md"
+								}
```
Comment (samoretc) [inline extensionsGallery.json]: Do I need to upload the manifest (package.json) and License as well?

File: extensionsGallery.json [hunk 5]
```diff
@@ -63,6 +63,59 @@
...
+							],
+							"properties": [
+								{
+									"key": "Microsoft.VisualStudio.Code.ExtensionDependencies",
+									"value": ""
```
Comment (samoretc) [inline extensionsGallery.json]: There's no dependent extension for intellicode, right? The mssql extension is apart of ADS, right?

File: extensionsGallery.json (hunk 1 above)
Comment (Charles-Gagnon) [reply extensionsGallery.json]: It's just an identifier. Use 57 for this and then make sure to update TotalCount to 47 below

File: extensionsGallery.json (hunk 2 above)
Comment (Charles-Gagnon) [reply extensionsGallery.json]: Feel free to put what values make sense for you, having the publisher be VisualStudioExptTeam seems fine

File: extensionsGallery.json (hunk 3 above)
Comment (Charles-Gagnon) [reply extensionsGallery.json]: You need to update the link to point to your VSIX. We can host that for you on our blob storage if you don't have a place yourself.

File: extensionsGallery.json (hunk 5 above)
Comment (Charles-Gagnon) [reply extensionsGallery.json]: Correct

File: extensionsGallery.json (hunk 4 above)
Comment (Charles-Gagnon) [reply extensionsGallery.json]: It would be good if you did. Is the extension not open source? I don't see anything in the github repo

Comment (samoretc) [reply extensionsGallery.json]: it's not open source

File: extensionsGallery.json (hunk 3 above)
Comment (samoretc) [reply extensionsGallery.json]: But what about the assetUri and fallBackAssetUri values. Are those correct?

Comment (samoretc) [reply extensionsGallery.json]: And, yes, if you could host the VSIX that would be fantastic! I will send you it through email

File: extensionsGallery.json (hunk 1 above)
Comment (samoretc) [reply extensionsGallery.json]: got it, thanks. I will also move this to the bottom of the JSON

File: extensionsGallery.json (hunk 2 above)
Comment (alanrenmsft) [reply extensionsGallery.json]: We don't have a strong preference here, as long as your team is fine with it. updating it in the future might cause some issue for the update scenario.

File: extensionsGallery.json (hunk 4 above)
Comment (alanrenmsft) [reply extensionsGallery.json]: I'd recommend to have a GITHUB repo for your extension to host the license, manifest and icons...

Comment (alanrenmsft) [reply extensionsGallery.json]: and also for users to open issues

File: extensionsGallery.json (hunk 3 above)
Comment (Charles-Gagnon) [reply extensionsGallery.json]: Yes, those values are fine for assetUri and fallBackAssetUri

**Comment by Charles-Gagnon** [comment]
You should also be adding this to https://github.com/microsoft/azuredatastudio/blob/release/extensions/extensionsGallery-insider.json

**Pull request merged**